docker-compose.yml

```diff
@@ -1,52 +1,74 @@
 version: '3'
 
 services:
+    composer-dev:
+        build:
+            context: .
+            dockerfile: Dockerfile.composer.dev
+        image: elifesciences/annotations_composer:${IMAGE_TAG}-dev
+        command: /bin/bash
+    composer-no-dev:
+        build:
+            context: .
+            dockerfile: Dockerfile.composer.no-dev
+        image: elifesciences/annotations_composer:${IMAGE_TAG}-no-dev
+        command: /bin/bash
     web:
         image: nginx:1.13.7
         ports:
             - "8080:80"
         volumes:
-            - ./config/nginx-default.conf:/etc/nginx/conf.d/default.conf
-            # can we avoid this by removing try_files?
-            - ./web:/srv/annotations/web
+            - ./.docker/nginx-default.conf:/etc/nginx/conf.d/default.conf
+            - web:/srv/annotations/web
         depends_on:
             - fpm
     fpm:
-        build: 
+        build:
             context: .
             dockerfile: Dockerfile.fpm
-        image: elifesciences/annotations_fpm
+            args:
+                composer_type: ${COMPOSER_TYPE}
+                image_tag: ${IMAGE_TAG}
+                dependencies_api_dummy: "${DEPENDENCIES_API_DUMMY}"
+                dependencies_hypothesis_dummy: "${DEPENDENCIES_HYPOTHESIS_DUMMY}"
+        image: elifesciences/annotations_fpm:${IMAGE_TAG}
         volumes:
-            - ./:/srv/annotations
-            - ./config/container.php:/srv/annotations/config.php
+            - ./.docker/config.php:/srv/annotations/config.php
             - logs:/srv/annotations/var/logs
+            - web:/srv/annotations/web
         env_file:
             - dev.env
         depends_on:
             - api_dummy
+            - composer-${COMPOSER_TYPE}
             - hypothesis_dummy
             - goaws
     cli:
-        build: 
+        build:
             context: .
             dockerfile: Dockerfile.cli
-        image: elifesciences/annotations_cli
+            args:
+                composer_type: ${COMPOSER_TYPE}
+                image_tag: ${IMAGE_TAG}
+                dependencies_api_dummy: "${DEPENDENCIES_API_DUMMY}"
+                dependencies_hypothesis_dummy: "${DEPENDENCIES_HYPOTHESIS_DUMMY}"
+        image: elifesciences/annotations_cli:${IMAGE_TAG}
         volumes:
-            - ./:/srv/annotations
```

Comment:
the mounting of the source folder as a volume is one of the main divergences between `dev` and `ci`. Using this file in `ci` correctly avoids mounting that folder (otherwise the content of the image is overwritten and the tests do not target it); but I thought local development needed these mounts

Reply:
Need to look at local development, especially with Composer.